### PR TITLE
remaining removal of isSystemMessages

### PR DESCRIPTION
**Pull request by dannimad**
# https://github.com/microsoft/FluidFramework/issues/9732

## Description

> Continuing with original PR https://github.com/microsoft/FluidFramework/pull/10430 work to get rid of isSystemMessages. 

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] My code follows the code style of this project.
-   [ ] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [ ] No

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.

## Testing

> -   Instructions for testing and validation of your code so the reviewer can follow those steps and validate code works as expected

## Any relevant logs or outputs

> -   Use this section to provide either screenshots or output of logs as code snippets

## Other information or known dependencies

> -   TODO that are to be done after this PR : remove isSystemMessage function

### CODE REVIEW

**Comment by dannimad** [comment]
not actually thinking to merge the pr but just for visibility purposes. Adding "attach" messageType fixed replay.spec.ts and "component" fixed opsOnReconnect.spec.ts

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -1721,8 +1721,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
         const local = this.clientId === message.clientId;
 
+        console.log("message is a ", message.type);
```
Comment (curtisman) [inline packages/loader/container-loader/src/container.ts:1724]: shouldn't leave console.log in the code.

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -1688,7 +1689,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
 
         // Forward non system messages to the loaded runtime for processing
-        if (!isSystemMessage(message)) {
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        if (isRuntimeMessage(message) || isLegacyRuntimeMessage(message)) {
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:1693]: I'd suggest fire telemetry event when we hit isLegacyRuntimeMessage() && !isRuntimeMessage
That way we can take a look if we ever hit them (in let's say 30-day timeframe). If we hit zero, I'd rather remove it from here.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -501,7 +501,7 @@ export function unpackRuntimeMessage(message: ISequencedDocumentMessage) {
             message.type = innerContents.type;
             message.contents = innerContents.contents;
         }
-        assert(isRuntimeMessage(message), 0x122 /* "Message to unpack is not proper runtime message" */);
+        assert(isLegacyRuntimeMessage(message), 0x122 /* "Message to unpack is not proper runtime message" */);
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:504]: I do not think naming here is correct. It's correct at loader level, where we should see zero such events (or very close to zero).
Here - pretty much every session will see such events. The better name for this layer is "unpacked runtime message".
I.e. even if check is exactly the same, I'd rather use correct terminology by using functions with same bodies.

That said, they should not have same bodies.
"Legacy" should not include such things as "rejoin" & "alias" - these are pretty recent additions, they were added long after runtime started to wrap everything via "op", so we should never see these ops at loader layer.
Not sure about other types, but possible that couple other types are in the same bucket.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 4]
```diff
@@ -1257,17 +1257,17 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this,
             (attachMsg) => this.submit(ContainerMessageType.Attach, attachMsg),
             (id: string, createParam: CreateChildSummarizerNodeParam) => (
-                    summarizeInternal: SummarizeInternalFn,
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1260]: I'd suggest splitting formatting PR from functional changes (if possible). Might be a bit late though (I guess it's more feedback for future changes)

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 3 above)
Comment (dannimad) [reply packages/runtime/container-runtime/src/containerRuntime.ts:504]: Probably we can just change name isLegacyRuntimeMessage to isUnpackedRuntimeMessage.

File: packages/loader/container-loader/src/connectionManager.ts [hunk 5]
```diff
@@ -780,7 +781,7 @@ export class ConnectionManager implements IConnectionManager {
 
     public sendMessages(messages: IDocumentMessage[]) {
         assert(this.connected, 0x2b4 /* "not connected on sending ops!" */);
-
+        // messages.forEach((message) => assert(isRuntimeMessage(message), "only client message allowed"));
```
Comment (curtisman) [inline packages/loader/container-loader/src/connectionManager.ts:784]: Why add this comment

**Review by curtisman — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 6]
```diff
@@ -340,11 +340,11 @@ export interface ISummaryRuntimeOptions {
...
     maxOpsSinceLastSummary?: number;
 
-     /**
-     * @deprecated - use `summaryConfigOverrides.summarizerClientElection` instead.
-     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
-     * This defaults to false (disabled) and must be explicitly set to true to enable.
-     */
+    /**
+    * @deprecated - use `summaryConfigOverrides.summarizerClientElection` instead.
+    * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:345]: nit: Indentation seems to be wonky here

File: packages/loader/container-loader/src/container.ts [hunk 7]
```diff
@@ -911,7 +912,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 throw newError;
             }
         },
-        { start: true, end: true, cancel: "generic" });
+            { start: true, end: true, cancel: "generic" });
```
Comment (kian-thompson) [inline packages/loader/container-loader/src/container.ts:915]: nit: Indentation

**Pull request merged**